src/provider/ticket.rs

```diff
@@ -70,16 +70,27 @@ impl Ticket {
     }
 
     /// The [`AuthToken`] for this ticket.
-    pub fn auth_token(&self) -> Option<AuthToken> {
-        self.auth_token.clone()
+    pub fn auth_token(&self) -> Option<&AuthToken> {
+        self.auth_token.as_ref()
     }
 
     /// The addresses on which the provider can be reached.
     ///
     /// This is guaranteed to be non-empty.
-    pub fn addrs(&self) -> &[SocketAddr] {
+    pub fn addrs(&self) -> &Vec<SocketAddr> {
         &self.addrs
     }
+
+    /// Get the contents of the ticket, consuming it.
+    pub fn destructure(self) -> (Hash, PeerId, Vec<SocketAddr>, Option<AuthToken>) {
```

Comment:
I like `into_parts` as a naming convention for these kind of functions

Reply:
sounds good :+1: